### PR TITLE
Initial words on provider UI plugins.

### PR DESCRIPTION
**Pull request by martinpovolny**
Initial words and examples on provider UI plugins.

### CODE REVIEW

**Comment by martinpovolny** [comment]
ping @himdel

File: ui/provider_plugin.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
+## Excending the UI with buttons, dialogs and executing Ansible Playbooks
+
+### Context
+We want to allow the provider authors to extend ManageIQ with Ansible playbooks
+and roles. ManageIQ core provides the capability to execute an Asible playbook
```
Comment (himdel) [inline ui/provider_plugin.md]: > Asible

Comment (martinpovolny) [reply ui/provider_plugin.md]: Fixed. Thx!

File: ui/provider_plugin.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+                                              }.to_json},
+              :klass => ApplicationHelper::Button::ButtonWithoutRbacCheck),
+        end
+      end
+    end
+  end
+end
+```
+
+Above is an exampe where provider `ManageIQ::Providers::Amazon` is extending existing toolbar `EmsCloudCenter`with a new group containing a single button.
```
Comment (himdel) [inline ui/provider_plugin.md]: example

File: ui/provider_plugin.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+We want to allow the provider authors to extend ManageIQ with Ansible playbooks
+and roles. ManageIQ core provides the capability to execute an Ansible playbook
+from the provider repository. The ManageIQ UI provides a way to call that
+playbook from a toolbar. Before calling the playbook additional parameters can
+be collected using a custom form.
+
+### Provider workflow
+
+ * Implement your Ansible playbooks and place them in the provider repo `manageiq-provider-xxx`.
+ * Implement your custom forms as React components or ManageIQ Dynamic (service) Dialog, place them in the provider repo.
```
Comment (himdel) [inline ui/provider_plugin.md]: maybe a note about `app/javascript`?

**Pull request merged**